vllm/v1/engine/core_client.py

```diff
@@ -982,7 +982,16 @@ async def run_engine_stats_update_task():
         resources.stats_update_task = asyncio.create_task(
             run_engine_stats_update_task())
 
-    def get_core_engine_for_request(self) -> CoreEngine:
+    def get_core_engine_for_request(self,
+                                    dp_rank: Optional[int] = None
+                                    ) -> CoreEngine:
+        if dp_rank is not None:
+            es = list(filter(lambda e: e.index == dp_rank, self.core_engines))
+            assert len(
+                es
+            ) == 1, f"Expected exactly one EngineCore with DP rank {dp_rank}"
+            return es[0]
```

Comment:
`core_engines` will be in rank order so you can just do:
```suggestion
            return self.core_engines[dp_rank]
```